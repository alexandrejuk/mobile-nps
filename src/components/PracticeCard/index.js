import React from 'react'
import PropTypes from 'prop-types'
import Title from '../Title'

import styles from './style.module.css'

import BannerSvg3 from '../../assets/images/banners/banner-3.svg'

const practicesImage = {
  BannerSvg3,
}

const PracticeCard = ({
  action,
  image,
  title,
  readTime,
}) => (
  <div
    onKeyPress={action}
    onClick={action}
    className={styles.card}
    tabIndex="0"
    role="button"
  >
    <div className={styles.imageContainer}>
      <img src={practicesImage[image]} alt="item card" />
    </div>
    <div className={styles.content}>
      <Title
        text={title}
        type="bold"
        size="medium"
      />
      <Title
        text={readTime}
        size="small"
        textAlign="right"
        type="mediumBold"
        color="secondary"
      />
    </div>
   </div>
)

PracticeCard.propTypes = {
  action: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  readTime: PropTypes.string,
}

PracticeCard.defaultProps = {
  readTime: null,
}

export default PracticeCard
