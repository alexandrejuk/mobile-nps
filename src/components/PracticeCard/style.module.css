@import '../../assets/css/colors.css';

.card {
  background: var(--color-white);
  border-radius: 8px;
  box-shadow: 0px 2.27027px 2.27027px rgba(0, 0, 0, 0.24), 0px 0px 2.27027px rgba(0, 0, 0, 0.12);
  max-height: 240px;
  text-align: center;
  width: 100%;
  flex: 1;
}

.card:active,
.card:focus {
  outline: none;
}

.imageContainer {
  background-color: var(--color-california);
  border-radius: 8px 8px 0 0;
  box-sizing: border-box;
  height: 131px;
  overflow: hidden;
  padding: 10px;
}

.imageContainer img {
  max-height: 131px;
  max-width: 320px;
  width: 100%;
}

.secondary {
  background-color: var(--color-cardinal);
}

.content {
  padding: 16px;
  min-height: 42px;
}

.content > :first-child {
  margin: 0;
}
