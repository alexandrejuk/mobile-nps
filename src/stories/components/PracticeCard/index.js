import React from 'react'
import { action } from '@storybook/addon-actions'
import { PracticeCard } from '../../../components'

export default {
  title: 'Components/PracticeCard',
}

const props = {
  action: action('card action!'),
  image: 'BannerSvg3',
  title: "Resumo dos Procedimentos para Auditoria TCP",
  readTime: '5-6 minutos de leitura',
}

export const Default = () => (
  <PracticeCard {...props} />
)
