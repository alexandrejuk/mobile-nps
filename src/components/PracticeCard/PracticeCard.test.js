import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'

import PracticeCard from './'

const props = {
  image: 'BannerSvg3',
  title: "Resumo dos Procedimentos para Auditoria TCP",
  readTime: '5-6 minutos de leitura',
}


describe('PracticeCard component', () => {
  it('should render correctly', () => {
  const callback = jest.fn()
    const { container, getByText, getByAltText } = render(
      <PracticeCard
        {...props}
        action={callback}
      />
    )

    const imageCard = getByAltText("item card")
    const SampleCard = getByText(props.title)
    expect(callback).not.toHaveBeenCalled()

    fireEvent.click(SampleCard)

    expect(SampleCard.innerHTML).toEqual(props.title)
    expect(imageCard.getAttribute('src')).toEqual('banner-3.svg')
    expect(callback).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
