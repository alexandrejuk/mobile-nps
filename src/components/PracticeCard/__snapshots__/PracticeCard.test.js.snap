// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PracticeCard component should render correctly 1`] = `
<div>
  <div
    class="card"
    role="button"
    tabindex="0"
  >
    <div
      class="imageContainer primary"
    >
      <img
        alt="item card"
        src="banner-3.svg"
      />
    </div>
    <div
      class="content"
    >
      <h1
        class="primary medium left title bold"
      >
        Resumo dos Procedimentos para Auditoria TCP
      </h1>
      <h1
        class="secondary small right title mediumBold"
      >
        5-6 minutos de leitura
      </h1>
    </div>
  </div>
</div>
`;
